io/io/src/TBufferFile.cxx

```diff
@@ -1937,10 +1944,13 @@ void TBufferFile::WriteArrayDouble32(const Double_t *d, Int_t n, TStreamerElemen
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Write array of n bools into the I/O buffer.
+/// \note Operation is silently skipped in case of underflow or overflow
```

Comment:
Silently not writing data is not an option (i.e. it will lead to undetected data loss).  Instead of return we ought to fail.  Given the current structure (no return value), the only option is to have a call `Fatal`.